app/models/status_pin.rb

```diff
@@ -17,11 +17,15 @@ class StatusPin < ApplicationRecord
 
   validates_with StatusPinValidator
 
-  after_destroy :invalidate_cleanup_info
+  after_destroy :invalidate_cleanup_info, if: %i(account_matches_status_account? account_local?)
 
-  def invalidate_cleanup_info
-    return unless status&.account_id == account_id && account.local?
+  delegate :local?, to: :account, prefix: true
 
+  def invalidate_cleanup_info
     account.statuses_cleanup_policy&.invalidate_last_inspected(status, :unpin)
   end
+
+  def account_matches_status_account?
+    status&.account_id == account_id
+  end
```

Comment:
I suppose these methods could be private?

Reply:
Yes they should -- original was not so I missed that. Added.